Reworks the migration sections to better showcase Hypershift (#2230)
timescale/docs: use-timescale/migration/about-hypershift.md

connection is interrupted. Additionally, when Hypershift is running, it holds a
single transaction open for the entire duration of the migration. This prevents
any automated vacuum tasks from running, which can cause a range of different
problems on a busy source database.
</Highlight>

## The hypershift command line tool

Hypershift is launched from the command prompt. You can provide a number of
different options to hypershift when you issue the command. This section
describes the various options.

You can also provide configuration to hypershift using a YAML configuration
file. For more information about constructing a hypershift configuration file,
see the
[hypershift configuration section][hypershift-config].

You can provide the source and target database URIs using either command line
flags, or the configuration file. If the information is provided in both,
hypershift uses the URIs provided in the command line flags, and ignores the
URIs in the configuration file.

To see all available configuration options for hypershift, use this command:

```bash
docker run -ti timescale/hypershift:0.5 --help
```

> Within the Docker container, use this command:
> 
> ```bash
> hypershift --help
> ```

Author: ```suggestion
```

---
title: About migrating to Timescale Cloud with hypershift
excerpt: Migrate an existing PostgreSQL database to Timescale Cloud in a single step
products: [cloud]
keywords: [data migration, Hypershift]
tags: [ingest, Hypershift, postgresql]
---

import HypershiftIntro from "versionContent/_partials/_hypershift-intro.mdx";

# About Hypershift

<HypershiftIntro />

## Migrating from a managed database

If you're migrating from an Amazon RDS database, the best option is to run
hypershift on an EC2 instance in the same availability zone as your RDS
instance.

<Highlight type="note">
Availability zones are assigned different names in different AWS
accounts. To make sure that you are using the same zone refer to
[AWS user guide](https://docs.aws.amazon.com/ram/latest/userguide/working-with-az-ids.html)

> </Highlight>

Reviewer: I don't think that we should include this. It's not actionable by the user, so only adds confusion.

Author: ```suggestion
```

stable network connection. Hypershift is not able to recover if the network
connection is interrupted. Additionally, when Hypershift is running, it holds a
single transaction open for the entire duration of the migration. This prevents
any automated vacuum tasks from running, which can cause a range of different
problems on a busy source database.
</Highlight>

## The hypershift command line tool

Hypershift is launched from the command prompt. You can provide a number of
different options to hypershift when you issue the command. This section
describes the various options.

You can also provide configuration to hypershift using a YAML configuration
file. For more information about constructing a hypershift configuration file,
see the
[hypershift configuration section][hypershift-config].

You can provide the source and target database URIs using either command line
flags, or the configuration file. If the information is provided in both,
hypershift uses the URIs provided in the command line flags, and ignores the
URIs in the configuration file.

To see all available configuration options for hypershift, use this command:

```bash

> docker run -ti timescale/hypershift:0.4 --help

Reviewer: ```suggestion
docker run -ti timescale/hypershift:0.5 --help
```

```

> Within the Docker container, use this command:

Reviewer: I'm not sure that showing the execution in the docker container is useful. I doubt that many users will expect to run it this way.

```bash

> ./hypershift --help

Reviewer: ```suggestion
hypershift --help
```

```

|Short command|Long command|Example use|Description|

> |-|-|-|-|

Reviewer: For completeness, this table is missing the following flags:

||--source-password|--source-password <PASSWORD>| Source database password|
||--target-password|--target-password <PASSWORD>| Target database password|

> |-C|--config-file|-C <CONFIG_FILE>|Path for hypershift configuration file|

Reviewer: ```suggestion
|-c|--config-file|-C <CONFIG_FILE>|Path for hypershift configuration file|
```

> |-S|--source-uri|-S <SOURCE_URI>|URI to the source database|

Reviewer: ```suggestion
|-s|--source|-s <SOURCE_URI>|URI to the source database|
```

> |-T|--target-uri|-T <TARGET_URI>|URI to the target database|

Reviewer: ```suggestion
|-t|--target|-t <TARGET_URI>|URI to the target database|
```

> ||--ignore-missing-time-index||Ignore hypertables that are missing a time index|

Reviewer: Correction: it doesn't ignore the hypertable, it ignores that the hypertable is missing an index.
```suggestion
||--ignore-missing-time-index||Ignore that hypertables may be missing a time index|
```

|-h|--help||Print hypershift help|
|-V|--version|Print hypershift version|

### Example hypershift commands

Migrate using the default configuration settings, without using a separate
configuration file, providing the source and target database URIs at the command
prompt:

```bash

> ./hypershift -S='host=localhost dbname=postgres port=5432 user=postgres password=source_password' \
> -T='host=localhost dbname=postgres port=5433 user=postgres password=target_password'

Reviewer: Aside from the arguments being wrong, I think that we should also do this as a docker container invocation
```suggestion
docker run -ti timescale/hypershift:0.5 -s='host=localhost dbname=postgres port=5432 user=postgres password=source_password' \
-t='host=localhost dbname=postgres port=5433 user=postgres password=target_password'
```

```

Migrate using a configuration file only:

```bash

> ./hypershift -C=config.yaml

Reviewer: Also here, I think that we should do it as a docker invocation
```suggestion
docker run -v $(pwd)/config.yaml:/config.yaml -ti timescale/hypershift:0.5 -c=config.yaml
```

```

Migrate using a configuration file, and provide source and target database URIs
at the command prompt:

```bash

> ./hypershift -C=config.yaml \
> -S='host=localhost dbname=postgres port=5432 user=postgres password=source_password' \
> -T='host=localhost dbname=postgres port=5433 user=postgres password=target_password'

Reviewer: ```suggestion
docker run -v $(pwd)/config.yaml:/config.yaml -ti timescale/hypershift:0.5 -c=config.yaml \
-s='host=localhost dbname=postgres port=5432 user=postgres password=source_password' \
-t='host=localhost dbname=postgres port=5433 user=postgres password=target_password'
```
